task-sdk/src/airflow/sdk/bases/operator.py

```diff
@@ -1607,6 +1615,216 @@ def resume_execution(self, next_method: str, next_kwargs: dict[str, Any] | None,
+
+        from airflow.models.dagrun import DagRun
+        from airflow.models.taskinstance import TaskInstance
+        from airflow.settings import Session
+        from airflow.utils.state import DagRunState
+        from airflow.utils.types import DagRunTriggeredByType, DagRunType
+
+        if session is None:
+            session = Session()
+
+        # Assertions for typing -- we need a dag, for this function, and when we have a DAG we are
+        # _guaranteed_ to have start_date (else we couldn't have been added to a DAG)
+        if TYPE_CHECKING:
+            from airflow.models.dag import DAG as SchedulerDAG
+
+            assert session
+            assert self.start_date
+
+            # TODO: Task-SDK: We need to set this to the scheduler DAG until we fully separate scheduling and
+            # definition code
+            assert isinstance(self.dag, SchedulerDAG)
+
+        start_date = pendulum.instance(start_date or self.start_date)
+        end_date = pendulum.instance(end_date or self.end_date or timezone.utcnow())
+
+        for info in self.dag.iter_dagrun_infos_between(start_date, end_date, align=False):
+            ignore_depends_on_past = info.logical_date == start_date and ignore_first_depends_on_past
+            try:
+                dag_run = session.scalars(
+                    select(DagRun).where(
+                        DagRun.dag_id == self.dag_id,
+                        DagRun.logical_date == info.logical_date,
+                    )
+                ).one()
+                ti = TaskInstance(self, run_id=dag_run.run_id)
+            except NoResultFound:
+                # This is _mostly_ only used in tests
+                dr = DagRun(
+                    dag_id=self.dag_id,
+                    run_id=DagRun.generate_run_id(
+                        run_type=DagRunType.MANUAL,
+                        logical_date=info.logical_date,
+                        run_after=info.run_after,
+                    ),
+                    run_type=DagRunType.MANUAL,
+                    logical_date=info.logical_date,
+                    data_interval=info.data_interval,
+                    run_after=info.run_after,
+                    triggered_by=DagRunTriggeredByType.TEST,
+                    state=DagRunState.RUNNING,
+                )
+                ti = TaskInstance(self, run_id=dr.run_id)
+                ti.dag_run = dr
+                session.add(dr)
+                session.flush()
+
+            ti.run(
+                mark_success=mark_success,
+                ignore_depends_on_past=ignore_depends_on_past,
+                wait_for_past_depends_before_skipping=wait_for_past_depends_before_skipping,
+                ignore_ti_state=ignore_ti_state,
+                test_mode=test_mode,
+                session=session,
+            )
+
+    # TODO (GH-52141): Either port this, or somehow fix the tests to remove this from the sdk.
+    def clear(
```

Comment:
Same `dry_run`, `clear`, `xcom_pull` and `xcom_push` shouldn't be needed anymore on BaseOperator.

Reply:
Let me test what fail without it in core: https://github.com/apache/airflow/pull/52888

Reply:
ok tests have been fixed as part of https://github.com/apache/airflow/pull/52888